New Footer

jamesmadson opened the pull request:
@ajswis, I recently worked on a new design for the footer and wanted to get this pull request started as more of a review until the text/copy is finalized. 
([https://invis.io/FJWMDOJ4H5D](https://invis.io/FJWMDOJ4H5D)).

Kirby is getting final approvals on the copy, but the styles and different breakpoints are setup. Can you take a look at the code (next week is fine) and let me know what you think? If anything could be optimized better, or consolidated?

@sawiggins Would you also take a look from a design perspective as a second set of eyes, also from a front-end perspective that the code is logical?

**Some things to note:**
- I've added 2 colors, I don't love the naming I chose, but not sure what else to name them ('blue-darkest' and 'blue-lightest')?
- Using same logo classes as the omni-bar/navbar, so wondering if we could break those out as global elements so they aren't duplicated twice?
- I'm using Font Awesome like the other footer, just making sure that all the Apps are doing the same.
- My `package-lock.json` file has updated to it, but I haven't pushed it to this branch, unless I should be? If not, should it be added to `.gitignore`?
- Is there anyone else that should be included on this PR?

Thanks!

ajswis commented on examples/footer/index.html:
```diff
@@ -8,59 +8,85 @@
+      <div class="container">
+        <div class="row">
+          <div class="col-6 col-lg-3 col-md-4 offset-xl-0 offset-lg-5 offset-md-4 offset-md-3 offset-3 footer-logo">
+            <a class="navbar-brand" href="https://stjude.cloud">
```
If it doesn't make it too complex, I would pull out the styles associated with the St. Jude Child logo and St. Jude Cloud title (`.navbar-brand` and `.sjc-title`) from `scss/_sjc-navbar.scss` [here](https://github.com/stjudecloud/theme/blob/f65a86f6d16e9f4b9b0fefae822a7eeaa1e0cb2c/scss/_sjc-navbar.scss#L11-L13) and [here](https://github.com/stjudecloud/theme/blob/f65a86f6d16e9f4b9b0fefae822a7eeaa1e0cb2c/scss/_sjc-navbar.scss#L77-L89) (leave [this](https://github.com/stjudecloud/theme/blob/f65a86f6d16e9f4b9b0fefae822a7eeaa1e0cb2c/scss/_sjc-navbar.scss#L228-L232) in the navbar as it is contextually important to the navbar). I would create a separate component out of it. Creating a new component for this de-duplicates some of the styling you've defined for the footer and adds some reusability.

We should also avoid word wrapping the brand if possible:
<img width="393" alt="Screen Shot 2020-03-27 at 5 27 25 PM" src="https://user-images.githubusercontent.com/2433649/77805652-7290ac80-7050-11ea-8034-57035a456e0e.png">

ajswis commented on examples/footer/index.html:
```diff
@@ -8,59 +8,85 @@
+                <li>
+                  <a target="_blank" rel="noopener noreferrer" href="https://www.stjude.org/legal/notice-of-non-discrimination.html" title="Notice of Non-Discrimination">Non-Discrimination</a>
+                </li>
+              </ul>
```
I'm not sure if it's intended, but the list bullets are showing on large displays. At the very least, the spacing should be uniform between each item.

<img width="560" alt="Screen Shot 2020-03-27 at 5 15 50 PM" src="https://user-images.githubusercontent.com/2433649/77805534-1cbc0480-7050-11ea-83b1-4828b82a365e.png">

Additionally, the text color changes between sizing breakpoints:

<img width="577" alt="Screen Shot 2020-03-27 at 5 15 59 PM" src="https://user-images.githubusercontent.com/2433649/77805554-2a718a00-7050-11ea-9264-95b9cfd5fc35.png">

sawiggins replied:
I like the bullets as delimiters but agree to equal space (and I'd say at least that of what's to the right of each bullet)

burnwatt replied:
I think it's definitely nitpicky, but I prefer the spacing from the institutional footer over the bullets.

![image](https://user-images.githubusercontent.com/48360246/77930607-b5c96600-7270-11ea-9d87-af83b7bf22b0.png)

jamesmadson replied:
I've fixed the spacing of the bullet points, and changed the links to remain the same color on all sizes.

ajswis commented:
The `package-lock.json` should not be `.gitignore`d. I think if you didn't make any dependency changes, then just don't commit your local modifications.

claymcleod commented on examples/footer/index.html:
```diff
@@ -8,59 +8,85 @@
+                <a href="https://www.stjude.org/about-st-jude.html">About</a>
+              </li>
+              <li>
+                <a href="https://stjude.org/contact-us.html">Contact</a>
```
Question to the group: is it more important that we have our own contact information at the bottom of the page? Many people will just naturally scroll here and click "Contact", which will point them to the general St. Jude contact page.

@jamesmadson, I think it may be worth swapping this out (I can't imagine many people would actually intend to contact the link described here).

sawiggins replied:
@claymcleod are you suggesting changing the link to [this one (which is currently in the footer)](https://hospital.stjude.org/apps/forms/fb/st-jude-cloud-contact/) or creating a new one that's hosted on St. Jude Cloud rather than the hospital's website? 

At minimum, @jamesmadson, I think the link should be swapped out for the one I linked to above, but from a UX perspective, I'm in favor of a contact us page on SJC depending on what our options are. Our own page would also be an opportunity to present a link to the docs (which is no longer in the footer).

claymcleod commented on examples/footer/index.html:
```diff
@@ -8,59 +8,85 @@
+            <h6>St. Jude Cloud Apps</h6>
+            <ul>
+              <li>
+                <a href="https://cstn.stjude.cloud">CSTN</a>
```
```suggestion
                <a href="https://pecan.stjude.cloud">Pediatric Cancer Portal (PeCan)</a>
```

claymcleod commented on examples/footer/index.html:
```diff
@@ -8,59 +8,85 @@
+                <a href="https://cstn.stjude.cloud">CSTN</a>
+              </li>
+              <li>
+                <a href="https://platform.stjude.cloud">Platform</a>
```
```suggestion
                <a href="https://platform.stjude.cloud">Genomics Platform</a>
```

claymcleod replied:
I would change the order to what will match in our blog post:

* Genomics Platform
* PeCan
* Visualization Community

sawiggins commented on examples/footer/index.html:
```diff
@@ -8,59 +8,85 @@
+            <h6>About St. Jude</h6>
+            <ul>
+              <li>
+                <a href="https://www.stjude.org">Home</a>
```
Call me a dumb user but "Home" in the footer (even under the sub-heading of "About St. Jude" albeit next to our SJC logo), I still missed that the link would be going to the hospital's homepage initially...

I think what I'm missing is the context that SJC is powered by the hospital (which has been an on-going conversation at the top-level too), and I fear that the "About St. Jude" header isn't strong enough. Is there an opportunity to make that connection in the footer before we add several links to the hospital's website?

Alternatively, how many links back to the hospital's website do we need, @claymcleod and @kirbyjbirch? [Together's ](https://together.stjude.org/en-us/)footer links are all internal.

burnwatt replied:
I think it's the fact that we have the cloud logo featured prominently before users are naturally arriving at those links. I too would expect the footer links to be cloud oriented. Even the social links are pointing to non-cloud specific accounts.

sawiggins commented on scss/_variables.scss:
```diff
@@ -35,7 +35,9 @@ $black: #000 !default;
 $blue-dark: #002856 !default;
+$blue-darkest: #002044 !default;
 $blue-light: #eff5fc !default;
+$blue-lightest: #9EC0E7 !default;
```
What if we called `$blue-dark` `$navy` then `$blue-darkest` became `$navy-dark` and `$blue-lightest` became `$navy-light` such that we could delineate between the base of blue (#1874dc) and navy?

ajswis replied:
If you make this change, to prevent breakage, keep `$blue-dark: $navy !default` so apps already using `$blue-dark` don't need an immediate update.

jamesmadson replied:
I like that solution @sawiggins. @ajswis Will make sure to keep `$blue-dark`.

sawiggins commented on examples/footer/index.html:
```diff
@@ -8,59 +8,85 @@
+            </ul>
+          </div>
+          <div class="col-md-3 col-6 offset-xl-0 offset-md-2 footer-apps">
+            <h6>St. Jude Cloud Apps</h6>
```
@jamesmadson do you have a vision for what adding an extra app will look like? I know we only have three now (which lines up nice with the other column) but could see that number growing in the not so distant future.

jamesmadson replied:
I think we'll address that when we come to it. As it sits right now, the columns would likely be ok with up to 6 apps listed. We'd maybe need to see how the apps are categorized (if at all) on the main SJ Cloud site.

ajswis commented:
@jamesmadson Just wanted to follow up, do you mind updating the react component footer or would you prefer someone else take a stab at it?

claymcleod commented:
> Overall I think it looks good. I think there's a bit of philosophical debate as to whether or not our footer content should be more cloud-centric vs institutional-centric (judging from comments @sawiggins and @claymcleod have made in their reviews), but that could definitely be addressed in a later iteration.

My two cents would definitely be cloud related.

jamesmadson commented:
@adthrasher Would probably be best for someone with React experience to take that implementation on. Thanks

@claymcleod I'm working with Kirby on some changes to the links and getting approval before finishing this up. Thanks all (CC @burnwatt & @sawiggins) for your input and patience while we work to wrap this up.

sawiggins commented on examples/footer/index.html:
```diff
@@ -11,52 +11,52 @@
               </li>
               <li>
-                <a href="https://stjude.org/contact-us.html">Contact</a>
+                <a href="https://stjude.cloud/terms-of-use" title="Contact">Contact</a>
```
I like that we are linking to SJC, but this link and anchor seem incongruent (also below).

jamesmadson commented on examples/footer/index.html:
```diff
@@ -8,59 +8,91 @@
+    <footer>
+      <div class="container">
+        <div class="row">
+          <div class="col-6 col-lg-3 col-md-4 offset-xl-0 offset-lg-5 offset-md-4 offset-md-3 offset-3">
```
Good catch, thanks!

ajswis commented:
@jamesmadson Are there any further changes you are intending for this pull request? I'd be happy to merge this if it is finished.

jamesmadson commented:
@ajswis Almost! We're just finalizing some items with legal. Just taking a bit longer than expected. Thanks for checking in on it.

ajswis commented:
@jamesmadson Great! Thanks! Just re-request a review when this is ready, then.

jamesmadson commented:
@ajswis OK! I think we're set. I updated the footer and the React footer since it was easy to follow the link code. I just added `Linking` in to the legal set of links, and removed the word `Policy` from it and `Privacy` to match them both to `Privacy (HIPAA)` and shorten them so it all fits on one line on large screens, AND reduces wording.

claymcleod commented:
Are we close on merging this in?

ajswis commented:
@claymcleod Yes.
